### PR TITLE
MM-27750 Add API's for in product notices feature

### PR DESCRIPTION
**Pull request by sudheerDev**
#### Summary
  * Add API's for in product notices
  * Add types for notices data
  * Add types for socket status reducer

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-27750

### CODE REVIEW

**Review by sbishel — APPROVED** [review]

**Review by Willyfrog — APPROVED** [review]

File: src/actions/teams.ts [hunk 1]
```diff
@@ -740,3 +740,23 @@ export function membersMinusGroupMembers(teamID: string, groupIDs: Array<string>
...
+        clientFunc: Client4.getInProductNotices,
+        params: [
+            teamId,
+            client,
+            clientVersion,
+        ],
+    });
+}
+
+export function updateNoticeAsViewed(noticeIds: string[]): ActionFunc {
```
Comment (reflog) [inline src/actions/teams.ts:755]: better name would be updateNoticesAsViewed since we accept an array

File: src/client/client4.ts [hunk 2]
```diff
@@ -3336,6 +3341,20 @@ export default class Client4 {
...
     };
 
+    getInProductNotices = (teamId: string, client: string, clientVersion: string) => {
+        return this.doFetch<ProductNotices>(
+            `${this.getNoticesRoute()}/${teamId}?client=${client}&clientVersion=${clientVersion}`,
+            {method: 'get'},
+        );
+    };
+
+    updateNoticeAsViewed = (noticeIds: string[]) => {
```
Comment (reflog) [inline src/client/client4.ts:3351]: same

**Review by reflog — APPROVED** [review]

File: src/actions/teams.ts (hunk 1 above)
Comment (sudheerDev) [reply src/actions/teams.ts]: Good call, Changed it

**Comment by sudheerDev** [comment]
QA review done on server PR,

**Pull request merged**